tooling/hibernate-gradle-plugin/hibernate-gradle-plugin.gradle

```diff
@@ -4,26 +4,83 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-apply plugin: 'groovy'
+plugins {
+	id "groovy"
 
-description = "Gradle plugin for integrating Hibernate functionality into your build"
+	id 'java-gradle-plugin'
+	id "com.gradle.plugin-publish" version "0.12.0"
+	id "nu.studer.credentials" version "2.1"
+
+	// for publishing snapshots
+	id 'maven-publish'
+	id 'org.hibernate.build.maven-repo-auth'
+
+	id 'idea'
+	id 'eclipse'
+}
+
+description = "Gradle plugin for integrating Hibernate aspects into your build"
 
 apply from: rootProject.file( 'gradle/published-java-module.gradle' )
-apply plugin: 'java-gradle-plugin'
-apply plugin: 'maven'
 
+ext {
+	pluginId = 'org.hibernate.orm'
+	pluginVersion = project.version
+
+	//noinspection GrUnresolvedAccess
+	publishKey = credentials.'hibernate.gradle.publish.key'
+	//noinspection GrUnresolvedAccess
+	publishSecret = credentials.'hibernate.gradle.publish.secret'
+
+	project.setProperty( 'gradle.publish.key', ext.publishKey )
+	project.setProperty( 'gradle.publish.secret', ext.publishSecret )
+}
 
 dependencies {
 	compile( project( ':hibernate-core' ) )
 	compile( libraries.jpa )
 	compile( libraries.javassist )
 	compile( libraries.byteBuddy )
-	compile gradleApi()
-	compile localGroovy()
 }
 
-tasks.withType( GroovyCompile ) {
-	options.encoding = 'UTF-8'
-	sourceCompatibility = project.baselineJavaVersion
-	targetCompatibility = project.baselineJavaVersion
+gradlePlugin {
+	plugins {
+		ormPlugin {
+			id = project.pluginId
+			implementationClass = 'org.hibernate.orm.tooling.gradle.HibernateOrmPlugin'
+		}
+	}
+}
+
+pluginBundle {
+	website = 'https://github.com/hibernate/hibernate-orm/tree/master/tooling/hibernate-gradle-plugin'
```

Comment:
The readme at this address is currently very... basic. Unless you intend to complete the readme (a link to a more complete documentation would probably be enough), maybe `https://docs.jboss.org/hibernate/stable/orm/userguide/html_single/Hibernate_User_Guide.html#BytecodeEnhancement-enhancement-gradle` would be a more helpful resource?

Reply:
Yes, this is a rough draft - but ultimately we will link to better documentation as well as provide better plugin specific doc at that URL.  I've made a todo to make sure that happens

Reply:
I improved the readme a little in the latest commit.  Still not done but better